devito/ir/iet/efunc.py

```diff
@@ -83,3 +93,196 @@ def make_efunc(name, iet, dynamic_parameters=None, retval='void', prefix='static
     """
     return ElementalFunction(name, iet, retval, derive_parameters(iet), prefix,
                              dynamic_parameters)
+
```

Comment:
Note for reviewers: most of this code here was just lifted from `passes/iet/orchestration.py`

Semantically, the changes are minimal and not particularly significant. The code overall looks much nicer now IMHO

Reply:
By 'lifted' do you mean transposed or copied? If it's the latter, I assume you considered and rejected creating an abstract class.

Reply:
by lifted I mean that I moved it (cut-paste), plus there are a few changes so that we generate C rather than C++, as we agreed. 

I don't know why you'd want an abstract class here, but the `Callable` hierarchy is already pretty deep